subprojects/execution/src/main/java/org/gradle/internal/execution/model/annotations/InputPropertyAnnotationHandler.java

```diff
@@ -92,9 +92,9 @@ private void validateNotFileType(PropertyMetadata propertyMetadata, TypeValidati
             validationContext.visitPropertyProblem(problem ->
                 problem
                     .forProperty(propertyMetadata.getPropertyName())
+                    .message(String.format("has @Input annotation used on property of type '%s'", ModelType.of(valueType).getDisplayName()))
```

Comment:
💭 Providing an overload of `message(String, Object...)` that would then forward to `String.format(String, Object...)` would be a nice enhancement to keep the calls to the Problem API a little cleaner.

Reply:
We already have it in the description, and we will introduce this to the `message(...)` methods soon!

Reply:
:thought_balloon: 
that is actually how the method looks like

`ProblemBuilderDefiningType message(String message, Object... args);`

We could apply it here :)

Reply:
Ah, it's description, which is not yet formatter friendly!